indico/web/client/js/react/forms/fields.jsx

```diff
@@ -10,17 +10,9 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import {Field, useFormState} from 'react-final-form';
 import {OnChange} from 'react-final-form-listeners';
-import {
-  Button,
-  Checkbox,
-  Dropdown,
-  Form,
-  Input,
-  Popup,
-  Radio,
-  TextArea,
-  Icon,
-} from 'semantic-ui-react';
+import {Button, Dropdown, Form, Input, Popup, Radio, TextArea, Icon} from 'semantic-ui-react';
+
+import Checkbox from 'indico/react/components/Checkbox';
```

Comment:
Really weird, when I change this to `import Checkbox from 'indico/react/components';` which should be pretty much equivalent since there's a re-report in components, I get this strange error/warning from webpack:

```
WARNING in chunk common [mini-css-extract-plugin]
Conflicting order. Following module has been added:
 * css ../../../node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[2].oneOf[2].use[1]!../../../node_modules/resolve-url-loader/index.js??ruleSet[1].rules[2].oneOf[2].use[2]!../../../node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[2].oneOf[2].use[3]!../../../node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[2].oneOf[2].use[4]!./js/react/components/principals/items.module.scss
despite it was not able to fulfill desired ordering with these modules:
 * css ../../../node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[2].oneOf[2].use[1]!../../../node_modules/resolve-url-loader/index.js??ruleSet[1].rules[2].oneOf[2].use[2]!../../../node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[2].oneOf[2].use[3]!../../../node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[2].oneOf[2].use[4]!./js/react/forms/fields.module.scss
   - couldn't fulfill desired order of chunk group(s) jquery, exports, module_events.abstracts, module_events.persons, module_events.registration, module_events.sessions, module_events.surveys, module_categories, module_categories.management, module_categories.calendar, module_events.display, module_events.header, module_receipts.admin, module_users.dashboard, module_users.profile_picture, module_users.favorites
   - while fulfilling desired order of chunk group(s) main, module_events.contributions, module_events.editing, module_events.management, module_events.papers, module_attachments, module_auth.signup, module_core, module_events.admin, module_events.roles, module_logs, module_rb, module_receipts, module_search, module_users.personal_data, module_users.data_export
```

(and a few more similar ones)

Do you have any idea which of your changes in this PR may be causing this? Because I have the feeling this may come back to haunt us in the future if some change is introducing an inconsistency w/ CSS ordering...

Reply:
So while I don't have ideas as to what would cause this or how to find out what causes it, my hypothesis is that we are artificially enforcing an alphabetical order in the components whereas the actual order of (CSS) dependencies is (should be) different. To solve this issue once and for all we have two options:

1. Never import CSS in JSX modules
2. Never create god modules that import everything

I'm very much in favor of the second option as it also helps improve tree-shaking.